libraries/wutdevoptab/devoptab_fs_statvfs.c

```diff
@@ -5,7 +5,49 @@ __wut_fs_statvfs(struct _reent *r,
                  const char *path,
                  struct statvfs *buf)
 {
-   //TODO: FSGetFileSystemInfo
-   r->_errno = ENOSYS;
-   return -1;
+   FSStatus status;
+   FSCmdBlock cmd;
+   uint64_t freeSpace;
+   FSStat driveStat;
+   memset(buf, 0, sizeof(struct statvfs));
+
+   char *fixedPath = __wut_fs_fixpath(r, path);
+   if (!fixedPath) {
+      return -1;
+   }
+
+   FSInitCmdBlock(&cmd);
+
+   status = FSGetFreeSpaceSize(__wut_devoptab_fs_client, &cmd, fixedPath, &freeSpace, FS_ERROR_FLAG_ALL);
+   if (status < 0) {
+      r->_errno = __wut_fs_translate_error(status);
+      return -1;
+   }
+
+   status = FSGetStat(__wut_devoptab_fs_client, &cmd, fixedPath, &driveStat, FS_ERROR_FLAG_ALL);
+   if (status < 0) {
+      r->_errno = __wut_fs_translate_error(status);
+      return -1;
+   }
+
+   // File system block size
+   buf->f_bsize = 512;
+   // Fundamental file system block size
+   buf->f_frsize = 512;
```

Comment:
Is this true for all filesystems?

Reply:
Plans have changed and this devoptab will no longer fulfill a double purpose where it might have to support reading the system drives too. Adding the code required to support it would add bloat unnecessarily.